synapse/federation/federation_server.py

```diff
@@ -376,10 +379,16 @@ async def on_state_ids_request(
         if not in_room:
             raise AuthError(403, "Host not in room.")
 
+        resp = await self._state_ids_resp_cache.wrap(
+            (room_id, event_id), self._on_state_ids_request_compute, room_id, event_id,
+        )
```

Comment:
The call above to `self._state_resp_cache` gets wrapped by `dict()`, which we don't do here. It doesn't seem necessary, but curious about the difference.

Reply:
the `dict` there was added in https://github.com/matrix-org/synapse/pull/6176/files. It looks like it's so that we can add a `room_version` property after it comes out of the cache, without accidentally modifying the cached response.

It is unclear to me why that implementation was chosen instead of adding the room_id in `_on_context_state_request_compute`.

In any case, I think it's unnecessary in the new code.